prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/migrationtoolkit/CreateApplicationTask.java

```diff
@@ -0,0 +1,92 @@
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * {@link CreateApplicationTask} Creates an application under the application inventory
+ * with the details of the source repository to scan.
+ */
+@Slf4j
+public class CreateApplicationTask extends BaseInfrastructureWorkFlowTask {
+
+	// The MTA application client to work with. If kept null then a client would be
+	// created using the context params serverURL and bearerToken during execution time.
+	// Not depending on the context params for the client details
+	// means flow authors can build the client object with their
+	// params and pass it is. This also prevents an invoker to pass the client
+	// details by ignoring them if a client already exists.
+	// This method is useful for testing as well.
+	protected MTAApplicationClient mtaClient;
+
+	public CreateApplicationTask() {
+	}
+
+	public CreateApplicationTask(URI serverURL, String bearerToken) {
+		mtaClient = new MTAClient(serverURL, bearerToken);
+	}
+
+	@Override
+	public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
+		return List.of(
+				WorkParameter.builder().key("applicationName").type(WorkParameterType.TEXT).optional(false)
+						.description("The application name. Can be generated from the repository name").build(),
+				WorkParameter.builder().key("repositoryURL").type(WorkParameterType.TEXT).optional(false)
+						.description("The application git repository URL. Can be generated from the repository name")
+						.build(),
+				WorkParameter.builder().key("serverURL").type(WorkParameterType.TEXT).optional(true).description(
+						"Base URL of the MTA instance - e.g https://mta-openshift-mta.app.clustername.clusterdomain")
+						.build(),
+				WorkParameter.builder().key("bearerToken").type(WorkParameterType.TEXT).optional(true)
+						.description("Bearer token to authenticate server requests").build());
+	}
+
+	/**
+	 * @param workContext optional context values: serverURL, and bearerToken for the
+	 * mtaClient.
+	 */
+	@Override
+	public WorkReport execute(WorkContext workContext) {
+		String appName, repo;
+		try {
+			appName = getOptionalParameterValue(workContext, "applicationName", "");
+			repo = getRequiredParameterValue(workContext, "repositoryURL");
+			if (mtaClient == null) {
+				var serverUrl = getOptionalParameterValue(workContext, "serverURL", null);
+				var bearerToken = getOptionalParameterValue(workContext, "bearerToken", null);
+				this.mtaClient = new MTAClient(URI.create(serverUrl), bearerToken);
+			}
+		}
+		catch (MissingParameterException e) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, e);
+		}
+
+		Result<App> result = mtaClient.create(new App(0, appName, new Repository("git", repo)));
+
+		if (result == null) {
+			// unexpected
+			return new DefaultWorkReport(WorkStatus.FAILED, new WorkContext(),
```

Comment:
yes please return workContext here.  future tasks inherited from this may want to get some origin workcontext info.

Reply:
> why not return the same `workContext` here? it might contain information relevant for the execution (e.g. the execution id, project id...)

three's no point at all to return a context. at least not in the current model. the aspect has a ref to the context anyhow

Reply:
@RichardW98 suggests that the context from the workreport is being used though... so I'll might change that after all

Reply:
@masayag  I just reviewed it again and it seems what returns here in workreport doesn't affect execution, as the workContext used in aspect and executionhandler is all referenced from: https://github.com/parodos-dev/parodos/blob/main/workflow-service/src/main/java/com/redhat/parodos/workflow/execution/aspect/WorkFlowExecutionAspect.java#LL85C17-L85C17,  which is not obtained from workreport

Reply:
I think it's best practice to return workContext, but let's fix it later